core/src/main/java/org/jdbi/v3/core/mapper/reflect/ConstructorInstanceFactory.java

```diff
@@ -27,19 +29,18 @@
 import static java.util.Objects.requireNonNull;
 
 class ConstructorInstanceFactory<T> extends InstanceFactory<T> {
-    private final Constructor<T> constructor;
+    private static final Map<Constructor, Type[]> CONSTRUCTOR_CACHE = new ConcurrentHashMap<>();
```

Comment:
This will prevent the JVM from unloading any class that the `ConstructorMapper` ever considers, since it keeps a static strong reference to the `Constructor`.

Would the change work just as well with a `synchronizedMap(new WeakHashMap<>())` or something like that? Then, the JVM would be able to unload classes since we do not strongly retain their Constructors.

Reply:
sounds good, will try all of these ideas out and update, thanks!